src/renderer/contexts/GlobalNodeState.tsx

```diff
@@ -755,6 +761,29 @@ export const GlobalProvider = memo(
             [modifyNode, schemata]
         );
 
+        const useInputSize = useCallback(
+            (
+                id: string,
+                inputId: InputId,
+                inputSize?: InputSize
+            ): readonly [Size, (size: Size) => void, () => void] => {
+                const currentSize = inputSize ? inputSize[inputId] : { width: 0, height: 0 };
+                const setInputSize = (size: Size) => {
+                    modifyNode(id, (old) => {
+                        const nodeCopy = copyNode(old);
+                        nodeCopy.data.inputSize = {
+                            ...nodeCopy.data.inputData,
```

Comment:
```suggestion
                            ...nodeCopy.data.inputSize,
```

Reply:
It's very strange that TS didn't catch this. For some reason, the type of the object comes out as `{}` (so any non-nullish value). This seems to be a bug in TS itself.

![image](https://user-images.githubusercontent.com/20878432/179350806-9e2c4273-7264-4f83-8736-3c4fbfe1c0f5.png)

Reply:
The problem seems to be that TS can't deal with branded types (e.g. `InputId`) as object keys. This seems to be a known issue.